sotodlib/toast/ops/sim_source.py

```diff
@@ -0,0 +1,694 @@
+import os
+import pickle
+import copy
+
+import traitlets
+
+import numpy as np
+
+from astropy import constants
+from astropy import units as u
+
+import ephem
+
+from scipy.interpolate import RectBivariateSpline
+
+from toast.timing import function_timer
+
+from toast import qarray as qa
+
+from toast.data import Data
+
+from toast.traits import trait_docs, Int, Unicode, Float, Instance, List, Bool, Quantity
+
+from toast.ops.operator import Operator
+from toast.instrument import Focalplane
+
+from toast.utils import Logger, Timer
+
+from toast.observation import default_values as defaults
+
+from ...coords.local import *
+
+
+
+def tb2s(tb, nu):
+    """ Convert blackbody temperature to spectral
+    radiance s_nu at frequency nu
+    Args:
+        tb: float or array
+            blackbody temperature, unit: Kelvin
+        nu: float or array (with same dimension as tb)
+            frequency where the spectral radiance is evaluated, unit: Hz
+    Return:
+        s_nu: same dimension as tb
+            spectral radiance s_nu, unit: W*sr−1*m−2*Hz−1
+    """
+    h = constants.h.value
+    c = constants.c.value
+    k_b = constants.k_B.value
+
+    x = h * nu / (k_b * tb)
+
+    return 2 * h * nu ** 3 / c ** 2 / (np.exp(x) - 1)
```

Comment:
We should probably make this a little more unit-correct... extracting a bunch of .value and then computing with them shows a hilarious disrespect for astropy.units.

Reply:
Ok I set them to just keep the units